receiver/elasticsearchreceiver/scraper.go

```diff
@@ -155,7 +170,16 @@ func (r *elasticsearchScraper) scrapeNodeMetrics(ctx context.Context, now pcommo
 		return
 	}
 
-	for _, info := range nodeStats.Nodes {
+	// Certain node metadata is not available from the /_nodes/stats endpoint. Therefore, we need to get this metadata
+	// from the /_nodes endpoint. The metadata may or may not be used depending on feature gates.
+	nodes, err := r.client.Nodes(ctx, r.cfg.Nodes)
+	if err != nil {
+		errs.AddPartial(26, err)
+		return
+	}
```

Comment:
Even though we will may want this behavior in the long term, it would be immediately breaking if there is something wrong. Let's use the feature gate to control whether we do any of this.

Reply:
I will use the same gate I introduced in this PR to keep the gating in sync.